test/cypress/integration/stripe-setup/vat-number-company-number/index_spec.js

```diff
@@ -0,0 +1,180 @@
+'use strict'
+
+// Local dependencies
+const commonStubs = require('../../../utils/common_stubs')
+const {
+  stubGetGatewayAccountStripeSetupSuccess,
+  stubGetStripeAccountSuccess
+} = require('./support')
+
+describe('Stripe setup: VAT number company number index page', () => {
+  const gatewayAccountId = 42
+  const userExternalId = 'userExternalId'
+
+  describe('Card gateway account', () => {
+    describe('when user is admin, account is Stripe and VAT number company number is not finished', () => {
+      beforeEach(() => {
+        cy.task('setupStubs', [
+          commonStubs.getUserStub(userExternalId, [gatewayAccountId]),
+          commonStubs.getGatewayAccountStub(gatewayAccountId, 'live', 'stripe'),
+          stubGetGatewayAccountStripeSetupSuccess(gatewayAccountId, false),
+          stubGetStripeAccountSuccess(gatewayAccountId, 'acct_123example123')
+        ])
+      })
+
+      it('should redirect to /vat-number page when pageData is empty', () => {
+        cy.setEncryptedCookies(userExternalId, gatewayAccountId, {})
+
+        cy.visit('/vat-number-company-number')
+
+        cy.location().should((location) => {
+          expect(location.pathname).to.eq('/vat-number-company-number/vat-number')
+        })
+      })
+
+      it('should redirect to /company-number page when pageData has vatNumber only', () => {
+        cy.setEncryptedCookies(userExternalId, gatewayAccountId, {
+          stripeSetup: {
+            vatNumberData: {}
```

Comment:
Might be more realistic to put something in here

Reply:
I put a blank object to show that redirect logic does not check on actual existence of the data, but it checks that `vatNumberData` exists as a key, to make it clear what logic expects